### PR TITLE
fix: user search by text instead of regex

### PR DESCRIPTION
**Pull request by nunogois**
## About the changes
Downgrades the new user search to a basic text search, which should be more intuitive for end-users in most cases.

### CODE REVIEW

**Review by sighphyre — APPROVED** [review]
LGTM

File: src/hooks/useUsersFilter.ts [hunk 1]
```diff
@@ -43,20 +43,12 @@ const filterUsersByQuery = (users: IUser[], filter: IUsersFilter): IUser[] => {
...
+        const search = filter.query?.toLowerCase();
+        return filterUserByText(user, search);
+    });
 };
 
-const filterUserByRegExp = (user: IUser, regExp: RegExp): boolean =>
-    regExp.test(user.name ?? '') ||
-    regExp.test(user.username ?? user.email ?? '');
+const filterUserByText = (user: IUser, search: string = ''): boolean =>
+    (user.name ?? '').toLowerCase().includes(search) ||
```
[the remark this answers is not in the document]
Comment (sighphyre) [reply src/hooks/useUsersFilter.ts:53]: @Tymek That seems to be handled in the calling function, might be better to move that into the filterUsersByText for clarity though

[the remark this answers is not in the document]
Comment (nunogois) [reply src/hooks/useUsersFilter.ts]: Please check: https://github.com/Unleash/unleash-frontend/pull/924/commits/1c183733344e1ac847e233a04dc06dcae3796833 - Hopefully it is clearer this way.

[the remark this answers is not in the document]
Comment (sighphyre) [reply src/hooks/useUsersFilter.ts]: Nice! Yeah looks good, thanks!

**Pull request merged**